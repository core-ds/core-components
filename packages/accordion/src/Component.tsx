import React, {
    AnchorHTMLAttributes,
    FC,
    KeyboardEvent,
    ReactNode,
    useCallback,
    useState,
} from 'react';
import cn from 'classnames';

import { Typography } from '@alfalab/core-components-typography';

import { DefaultControlIcon } from './components';
import { useRecalculateContentHeight } from './hooks';
import { ControlPosition } from './typings';

import styles from './index.module.css';

export type AccordionProps = {
    /**
     * Состояние компонента
     */
    expanded?: boolean;

    /**
     * Элемент заголовка
     */
    header: ReactNode;

    /**
     * Слот для элемента управления
     */
    control?: ReactNode;

    /**
     * Указывает компоненту - где будет размещен control
     */
    controlPosition?: ControlPosition;

    /**
     * Начальное состояние uncontrolled компонента
     */
    defaultExpanded?: boolean;

    /**
     * Основной элемент для отображения содержимого
     */
    children?: ReactNode;

    /**
     * Дополнительный класс обертки
     */
    className?: string;

    /**
     * Дополнительный класс для контейнера с заголовком
     */
    containerClassName?: string;

    /**
     * Дополнительный класс для header
     */
    headerClassName?: string;

    /**
     * Дополнительный класс для control
     */
    controlClassName?: string;

    /**
     * Дополнительный класс для body
     */
    bodyClassName?: string;

    /**
     * Обработчик смены состояний `expanded`
     */
    onExpandedChange?: (expanded: boolean) => void;

    /**
     * Идентификатор для систем автоматизированного тестирования
     */
    dataTestId?: string;
} & AnchorHTMLAttributes<HTMLDivElement>;

export const Accordion: FC<AccordionProps> = ({
    expanded,
    defaultExpanded = false,
    header,
    control,
    controlPosition = 'end',
    children,
    className,
    containerClassName,
    headerClassName,
    controlClassName,
    bodyClassName,
    onExpandedChange,
    dataTestId,
    ...rest
}) => {
    const uncontrolled = expanded === undefined;
    const [expandedState, setExpanded] = useState(uncontrolled ? defaultExpanded : expanded);
    const isExpanded = uncontrolled ? expandedState : expanded;

    const isStartPosition = controlPosition === 'start';

    const { contentRef, contentCaseRef } = useRecalculateContentHeight(isExpanded);

    const controlContent =
        control === undefined ? (
            <DefaultControlIcon expanded={isExpanded} startPosition={isStartPosition} />
        ) : (
            control
        );

    const headerContent =
        typeof header === 'string' ? (
            <Typography.Text view='primary-large' weight='medium'>
                {header}
            </Typography.Text>
        ) : (
            header
        );

    const bodyContent =
        typeof children === 'string' ? (
            <Typography.Text view='primary-medium'>{children}</Typography.Text>
        ) : (
            children
        );

    const handleExpandedChange = useCallback(() => {
        if (uncontrolled) {
            setExpanded(!isExpanded);
        }

        onExpandedChange?.(!isExpanded);
    }, [isExpanded, onExpandedChange, uncontrolled]);

    const handleKeyDown = (event: KeyboardEvent) => {
        if (event.key === 'Enter') {
            handleExpandedChange();
        }
    };

    return (
        <div {...rest} data-test-id={dataTestId} className={cn(styles.accordion, className)}>
            <div
                role='button'
                tabIndex={0}
                aria-expanded={isExpanded}
                onClick={handleExpandedChange}
                onKeyDown={handleKeyDown}
                className={cn(styles.container, containerClassName)}
            >
                <div
                    className={cn(styles.header, headerClassName, {
                        [styles.endPosition]: isStartPosition,
                    })}
                >
                    {headerContent}
                </div>

                <div
                    className={cn(styles.control, controlClassName, {
                        [styles.startPosition]: isStartPosition,
                    })}
                >
                    {controlContent}
                </div>
            </div>

            <div
                ref={contentRef}
                className={cn(styles.body, bodyClassName, { [styles.expandedBody]: isExpanded })}
            >
                <div className={cn(styles.bodyContent)} ref={contentCaseRef}>
                    {bodyContent}
                </div>
            </div>
        </div>
    );
};
