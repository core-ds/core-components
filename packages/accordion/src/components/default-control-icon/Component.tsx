import React, { FC } from 'react';
import cn from 'classnames';

import { ChevronDownMIcon } from '@alfalab/icons-glyph/ChevronDownMIcon';

import styles from './index.module.css';

type Props = {
    expanded: boolean;
    startPosition: boolean;
};
export const DefaultControlIcon: FC<Props> = ({ expanded, startPosition }) => (
    <ChevronDownMIcon
        className={cn(styles.icon, {
            [styles.expanded]: expanded,
            [styles.startPosition]: startPosition,
        })}
    />
);
