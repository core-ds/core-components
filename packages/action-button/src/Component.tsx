import React, {
    AnchorHTMLAttributes,
    ButtonHTMLAttributes,
    forwardRef,
    Fragment,
    useRef,
} from 'react';
import mergeRefs from 'react-merge-refs';
import cn from 'classnames';

import { Loader } from '@alfalab/core-components-loader';
import { useFocus } from '@alfalab/hooks';

import { useLoader } from './hooks';

import defaultColors from './default.module.css';
import styles from './index.module.css';
import invertedColors from './inverted.module.css';
import staticColors from './static.module.css';

const colorStyles = {
    default: defaultColors,
    inverted: invertedColors,
    static: staticColors,
};

/**
 * Минимальное время отображения лоадера - 500мс,
 * чтобы при быстрых ответах от сервера кнопка не «моргала».
 */
const LOADER_MIN_DISPLAY_INTERVAL = 500;

type Colors = 'default' | 'inverted' | 'static';

type ComponentProps = {
    /**
     * Иконка кнопки
     */
    icon: React.ReactNode;

    /**
     *  Размер кнопки
     */
    size?: 's';

    /**
     * Тип кнопки
     */
    view?: 'primary' | 'secondary';

    /**
     * Дополнительный класс
     */
    className?: string;

    /**
     * Дополнительный класс для обертки иконки
     */
    iconWrapperClassName?: string;

    /**
     * Значение href для ссылки
     */
    href?: string;

    /**
     * Заблокировать кнопку
     */
    disabled?: boolean;

    /**
     * Показать лоадер
     */
    loading?: boolean;

    /**
     * Id компонента для тестов
     */
    dataTestId?: string;

    /**
     * Палитра, в контексте которой используется кнопка
     */
    colors?: Colors;
};

type AnchorProps = ComponentProps & AnchorHTMLAttributes<HTMLAnchorElement>;
type ButtonProps = ComponentProps & ButtonHTMLAttributes<HTMLButtonElement>;
export type ActionButtonProps = Partial<AnchorProps | ButtonProps>;

export const ActionButton = forwardRef<HTMLAnchorElement | HTMLButtonElement, ActionButtonProps>(
    (
        {
            className,
            icon,
            children,
            href,
            size = 's',
            view = 'primary',
            type = 'button',
            iconWrapperClassName,
            disabled,
            loading,
            dataTestId,
            colors = 'default',
            ...rest
        },
        ref,
    ) => {
        const componentRef = useRef<HTMLElement>(null);

        const [focused] = useFocus(componentRef, 'keyboard');

        const { showLoader } = useLoader(!!loading, LOADER_MIN_DISPLAY_INTERVAL);

        const componentProps = {
            className: cn(
                styles.component,
                colorStyles[colors][view],
                styles[size],
                {
                    [styles.focused]: focused,
                    [styles.disabled]: disabled,
                    [styles.loading]: showLoader,
                },
                className,
            ),
            'data-test-id': dataTestId,
        };

        const buttonChildren = (
            <Fragment>
                <span
                    role='img'
                    className={cn(
                        styles.iconWrapper,
                        colorStyles[colors].iconWrapper,
                        styles[size],
                        iconWrapperClassName,
                    )}
                >
                    {showLoader ? <Loader dataTestId='loader' /> : icon}
                </span>
                <span className={styles.label}>{children}</span>
            </Fragment>
        );

        if (href) {
            return (
                <a
                    role='button'
                    ref={mergeRefs([componentRef, ref])}
                    href={href}
                    aria-disabled={disabled || loading}
                    {...componentProps}
                    {...(rest as AnchorHTMLAttributes<HTMLAnchorElement>)}
                >
                    {buttonChildren}
                </a>
            );
        }

        return (
            <button
                ref={mergeRefs([componentRef, ref])}
                // eslint-disable-next-line react/button-has-type
                type={type as ButtonHTMLAttributes<HTMLButtonElement>['type']}
                disabled={disabled || loading}
                {...componentProps}
                {...(rest as ButtonHTMLAttributes<HTMLButtonElement>)}
            >
                {buttonChildren}
            </button>
        );
    },
);
