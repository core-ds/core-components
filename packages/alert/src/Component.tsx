import React, { FC } from 'react';
import cn from 'classnames';

import { type Plate, type PlateProps } from '@alfalab/core-components-plate';
import { type PlateDesktop } from '@alfalab/core-components-plate/desktop';
import { type PlateMobile } from '@alfalab/core-components-plate/mobile';

import styles from './index.module.css';

export type AlertProps = Omit<PlateProps, 'foldable' | 'defaultFolded' | 'leftAddons'>;

type AlertPrivateProps = {
    Plate: typeof Plate | typeof PlateDesktop | typeof PlateMobile;
};

export const AlertBase: FC<AlertProps & AlertPrivateProps> = ({
    className,
    title,
    Plate,
    ...restProps
}) => (
    <Plate
        className={cn(styles.component, className)}
        title={title ? <span className={styles.title}>{title}</span> : null}
        rounded={false}
        limitContentWidth={false}
        {...restProps}
    />
);
