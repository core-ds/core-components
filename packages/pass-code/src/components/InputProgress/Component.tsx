/* eslint-disable react/no-array-index-key */
import React from 'react';
import { CSSTransition } from 'react-transition-group';
import cn from 'classnames';

import { getDataTestId } from '../../../../utils';

import styles from './index.module.css';
import transitions from './transitions.module.css';

export type InputProgressProps = {
    maxCodeLength: number;
    codeLength?: number;
    error: boolean;
    value?: string;
    dataTestId?: string;
};

const TRANSITION_DURATION = 150;

export const InputProgress: React.FC<InputProgressProps> = ({
    value = '',
    maxCodeLength,
    codeLength,
    error,
    dataTestId,
}) => (
    <div
        className={cn(styles.component, { [transitions.shake]: error })}
        data-test-id={getDataTestId(dataTestId, 'input-progress')}
    >
        {codeLength
            ? new Array(codeLength).fill(null).map((_, i) => {
                  const filled = Boolean(value[i]);

                  return (
                      <div
                          key={i}
                          className={cn(styles.dot, {
                              [styles.error]: filled && error,
                              [styles.filled]: filled,
                          })}
                      />
                  );
              })
            : new Array(maxCodeLength).fill(null).map((_, i) => (
                  <CSSTransition
                      key={i}
                      in={Boolean(value[i])}
                      timeout={TRANSITION_DURATION}
                      classNames={transitions}
                      unmountOnExit={true}
                  >
                      <div className={cn(styles.dot, styles.filled, { [styles.error]: error })} />
                  </CSSTransition>
              ))}
    </div>
);
