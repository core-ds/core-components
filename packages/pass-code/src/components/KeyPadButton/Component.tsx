import React, { ReactNode } from 'react';
import cn from 'classnames';

import { Button, ButtonProps } from '@alfalab/core-components-button';

import styles from './index.module.css';

export type KeyPadButtonProps<T> = {
    /**
     * Вид кнопки.
     */
    view: ButtonProps['view'];

    /**
     * Значение.
     */
    children: T;

    /**
     * Дополнительный класс.
     */
    className?: string;

    /**
     * Дополнительный класс кнопки.
     */
    buttonClassName?: string;

    /**
     * Коллбэк нажатия на кнопку.
     */
    onClick?: (payload: T) => void;
};

export function KeyPadButton<T extends ReactNode>({
    children,
    onClick,
    className,
    view = 'secondary',
    buttonClassName,
}: KeyPadButtonProps<T>) {
    return (
        <div className={cn(styles.component, className)}>
            <Button
                className={cn(styles.button, styles[view], buttonClassName)}
                view={view}
                onClick={() => onClick?.(children)}
            >
                {children}
            </Button>
        </div>
    );
}
