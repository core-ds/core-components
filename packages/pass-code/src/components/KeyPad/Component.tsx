import React, { ReactNode } from 'react';

import { BackspaceXxlIcon } from '@alfalab/icons-glyph/BackspaceXxlIcon';

import { getDataTestId } from '../../../../utils';
import { KeyPadButton } from '../KeyPadButton';

import styles from './index.module.css';

export type KeyPadProps = {
    /**
     * Показать кнопку "очистить".
     */
    showClear: boolean;

    /**
     * Слот слева.
     */
    leftAddons?: ReactNode;

    /**
     * Слот справа.
     */
    rightAddons?: ReactNode;

    /**
     * Идентификатор для автоматизированного тестирования.
     */
    dataTestId?: string;

    /**
     * Коллбэк нажатия на кнопку.
     */
    onClick: (digit: number) => void;

    /**
     * Коллбэк очистки кода.
     */
    onClear: () => void;
};

const CELL_COUNT = 12;
const CELLS = new Array(CELL_COUNT).fill(null).map((_, i) => {
    if (i === 10) return 0;

    return i + 1;
});

export const KeyPad: React.FC<KeyPadProps> = ({
    leftAddons,
    rightAddons,
    onClick,
    onClear,
    showClear,
    dataTestId,
}) => (
    <div className={styles.component} data-test-id={getDataTestId(dataTestId, 'keypad')}>
        {CELLS.map((digit, i) => {
            if (i === 9) {
                if (leftAddons) return leftAddons;

                return <div key='left-addons-empty' />;
            }

            if (i === 11) {
                if (showClear) {
                    return (
                        <KeyPadButton key='clear-btn' onClick={onClear} view='ghost'>
                            <BackspaceXxlIcon />
                        </KeyPadButton>
                    );
                }

                if (rightAddons) return rightAddons;

                return <div key='right-addon-empty' />;
            }

            return (
                <KeyPadButton
                    key={digit}
                    onClick={onClick}
                    view='secondary'
                    buttonClassName={styles.digit}
                >
                    {digit}
                </KeyPadButton>
            );
        })}
    </div>
);
