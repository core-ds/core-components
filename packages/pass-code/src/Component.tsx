import React, { forwardRef, ReactNode } from 'react';
import cn from 'classnames';

import { Gap } from '@alfalab/core-components-gap';

import { getDataTestId } from '../../utils';

import { InputProgress } from './components/InputProgress';
import { KeyPad } from './components/KeyPad';

import styles from './index.module.css';

export type BasePassCodeProps = {
    /**
     * Код.
     */
    value: string;

    /**
     * Обработчик изменения кода.
     */
    onChange: (code: string) => void;

    /**
     * Дополнительный класс.
     */
    className?: string;

    /**
     * Отображение ошибки
     */
    error?: ReactNode | boolean;

    /**
     * Слот слева.
     */
    leftAddons?: ReactNode;

    /**
     * Слот справа.
     */
    rightAddons?: ReactNode;

    /**
     * Идентификатор для систем автоматизированного тестирования.
     */
    dataTestId?: string;
};

type PropsWithUnknownLen = {
    codeLength?: never;

    /**
     * Максимально возможная длина кода.
     * @default 10
     */
    maxCodeLength?: number;
};

type PropsWithLen = {
    maxCodeLength?: never;

    /**
     * Длина кода.
     */
    codeLength?: number;
};

type PassCodeProps = BasePassCodeProps & (PropsWithLen | PropsWithUnknownLen);

export const PassCode = forwardRef<HTMLDivElement, PassCodeProps>(
    (
        {
            value = '',
            dataTestId,
            className,
            leftAddons,
            rightAddons,
            error,
            onChange,
            maxCodeLength = 10,
            codeLength,
        },
        ref,
    ) => {
        const passwordLen = codeLength || maxCodeLength;

        const handleChange = (digit: number) => {
            const newValue = value.concat(digit.toString());

            if (newValue.length <= passwordLen) {
                onChange?.(newValue);
            }
        };

        const handleClear = () => {
            if (value.length > 0) {
                onChange?.(value?.slice(0, -1));
            }
        };

        return (
            <div
                className={cn(styles.component, className)}
                ref={ref}
                data-test-id={getDataTestId(dataTestId, 'wrapper')}
            >
                <div className={styles.error} data-test-id={getDataTestId(dataTestId, 'error')}>
                    {error}
                </div>

                <Gap size='m' />

                <InputProgress
                    dataTestId={dataTestId}
                    value={value}
                    maxCodeLength={maxCodeLength}
                    codeLength={codeLength}
                    error={Boolean(error)}
                />

                <Gap size='4xl' />

                <KeyPad
                    dataTestId={dataTestId}
                    leftAddons={leftAddons}
                    rightAddons={rightAddons}
                    onClick={handleChange}
                    onClear={handleClear}
                    showClear={Boolean(value)}
                />
            </div>
        );
    },
);
