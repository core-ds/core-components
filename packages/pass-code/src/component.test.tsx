import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import { PassCode } from './Component';

jest.mock('react-transition-group', () => ({
    CSSTransition: jest.fn((props) => (props.unmountOnExit ? null : props.children)),
}));

describe('PassCode', () => {
    describe('Snapshot tests', () => {
        it('should match snapshot with code length', () => {
            const { container } = render(<PassCode value='' onChange={jest.fn} codeLength={4} />);

            expect(container).toMatchSnapshot();
        });

        it('should match snapshot with unknown code length', () => {
            const { container } = render(<PassCode value='' onChange={jest.fn} />);

            expect(container).toMatchSnapshot();
        });

        it('should match snapshot with error prop', () => {
            const { container } = render(
                <PassCode value='1234' error='Error message' onChange={jest.fn} />,
            );

            expect(container).toMatchSnapshot();
        });
    });

    describe('Callbacks tests', () => {
        it('should call `onChange` prop with new value', () => {
            const currentValue = '123';
            const pressedDigit = '4';

            const cb = jest.fn();

            const { getByText } = render(<PassCode value={currentValue} onChange={cb} />);

            const button = getByText(pressedDigit).parentNode as HTMLButtonElement;

            fireEvent.click(button);

            expect(cb).toBeCalledWith(currentValue + pressedDigit);
        });
    });

    describe('attributes tests', () => {
        it('should set `data-test-id` attribute', () => {
            const dataTestId = 'test-id';
            const { getByTestId } = render(
                <PassCode dataTestId={dataTestId} value='' onChange={jest.fn} />,
            );

            expect(getByTestId(dataTestId + '-wrapper')).toBeInTheDocument();
        });

        it('should set `className` class', () => {
            const className = 'test-class';
            const { container } = render(
                <PassCode className={className} value='' onChange={jest.fn} />,
            );

            expect(container.firstElementChild).toHaveClass(className);
        });

        it('should shown error message', () => {
            const dataTestId = 'pass-code';
            const errorMessage = 'Error message';
            const { getByText, getByTestId } = render(
                <PassCode
                    error={errorMessage}
                    value='1234'
                    codeLength={4}
                    onChange={jest.fn}
                    dataTestId={dataTestId}
                />,
            );

            const dots = getByTestId(dataTestId + '-input-progress').childNodes;

            dots.forEach((dot) => expect(dot).toHaveClass('error'));
            expect(getByText(errorMessage)).toBeInTheDocument();
        });

        it('should render left addons', () => {
            const leftAddonText = 'Left addon text';
            const { container, getByText } = render(
                <PassCode value='' onChange={jest.fn} leftAddons={leftAddonText} />,
            );

            expect(container.firstElementChild).toContainElement(getByText(leftAddonText));
        });

        it('should not render any dots with maxCodeLength and empty value props', () => {
            const dataTestId = 'pass-code';
            const { getByTestId } = render(
                <PassCode value='' onChange={jest.fn} maxCodeLength={8} dataTestId={dataTestId} />,
            );

            expect(getByTestId(dataTestId + '-input-progress').childNodes).toHaveLength(0);
        });

        it('should display four dots with codeLength equal to four', () => {
            const dataTestId = 'pass-code';
            const { getByTestId } = render(
                <PassCode value='' onChange={jest.fn} codeLength={4} dataTestId={dataTestId} />,
            );

            expect(getByTestId(dataTestId + '-input-progress').childNodes).toHaveLength(4);
        });

        it('should filled two dots with codeLength equal to four and value length equal to two', () => {
            const dataTestId = 'pass-code';
            const { getByTestId } = render(
                <PassCode value='12' onChange={jest.fn} codeLength={4} dataTestId={dataTestId} />,
            );

            const dots = getByTestId(dataTestId + '-input-progress').childNodes;

            expect(dots[0]).toHaveClass('filled');
            expect(dots[1]).toHaveClass('filled');
            expect(dots[2]).not.toHaveClass('filled');
            expect(dots[3]).not.toHaveClass('filled');
        });
    });

    it('should unmount without errors', () => {
        const { unmount } = render(<PassCode value='' onChange={jest.fn} />);

        expect(unmount).not.toThrowError();
    });
});
