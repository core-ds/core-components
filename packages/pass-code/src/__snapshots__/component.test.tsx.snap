// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PassCode Snapshot tests should match snapshot with code length 1`] = `
<div>
  <div
    class="component"
  >
    <div
      class="error"
    />
    <div
      class="gap vertical"
      data-gap-size="m"
    />
    <div
      class="component"
    >
      <div
        class="dot"
      />
      <div
        class="dot"
      />
      <div
        class="dot"
      />
      <div
        class="dot"
      />
    </div>
    <div
      class="gap vertical"
      data-gap-size="4xl"
    />
    <div
      class="component"
    >
      <div
        class="component"
      >
        <button
          class="component secondary m component secondary button secondary digit"
          type="button"
        >
          <span
            class="text stretchText"
          >
            1
          </span>
        </button>
      </div>
      <div
        class="component"
      >
        <button
          class="component secondary m component secondary button secondary digit"
          type="button"
        >
          <span
            class="text stretchText"
          >
            2
          </span>
        </button>
      </div>
      <div
        class="component"
      >
        <button
          class="component secondary m component secondary button secondary digit"
          type="button"
        >
          <span
            class="text stretchText"
          >
            3
          </span>
        </button>
      </div>
      <div
        class="component"
      >
        <button
          class="component secondary m component secondary button secondary digit"
          type="button"
        >
          <span
            class="text stretchText"
          >
            4
          </span>
        </button>
      </div>
      <div
        class="component"
      >
        <button
          class="component secondary m component secondary button secondary digit"
          type="button"
        >
          <span
            class="text stretchText"
          >
            5
          </span>
        </button>
      </div>
      <div
        class="component"
      >
        <button
          class="component secondary m component secondary button secondary digit"
          type="button"
        >
          <span
            class="text stretchText"
          >
            6
          </span>
        </button>
      </div>
      <div
        class="component"
      >
        <button
          class="component secondary m component secondary button secondary digit"
          type="button"
        >
          <span
            class="text stretchText"
          >
            7
          </span>
        </button>
      </div>
      <div
        class="component"
      >
        <button
          class="component secondary m component secondary button secondary digit"
          type="button"
        >
          <span
            class="text stretchText"
          >
            8
          </span>
        </button>
      </div>
      <div
        class="component"
      >
        <button
          class="component secondary m component secondary button secondary digit"
          type="button"
        >
          <span
            class="text stretchText"
          >
            9
          </span>
        </button>
      </div>
      <div />
      <div
        class="component"
      >
        <button
          class="component secondary m component secondary iconOnly button secondary digit"
          type="button"
        >
          0
        </button>
      </div>
      <div />
    </div>
  </div>
</div>
`;

exports[`PassCode Snapshot tests should match snapshot with error prop 1`] = `
<div>
  <div
    class="component"
  >
    <div
      class="error"
    >
      Error message
    </div>
    <div
      class="gap vertical"
      data-gap-size="m"
    />
    <div
      class="component shake"
    />
    <div
      class="gap vertical"
      data-gap-size="4xl"
    />
    <div
      class="component"
    >
      <div
        class="component"
      >
        <button
          class="component secondary m component secondary button secondary digit"
          type="button"
        >
          <span
            class="text stretchText"
          >
            1
          </span>
        </button>
      </div>
      <div
        class="component"
      >
        <button
          class="component secondary m component secondary button secondary digit"
          type="button"
        >
          <span
            class="text stretchText"
          >
            2
          </span>
        </button>
      </div>
      <div
        class="component"
      >
        <button
          class="component secondary m component secondary button secondary digit"
          type="button"
        >
          <span
            class="text stretchText"
          >
            3
          </span>
        </button>
      </div>
      <div
        class="component"
      >
        <button
          class="component secondary m component secondary button secondary digit"
          type="button"
        >
          <span
            class="text stretchText"
          >
            4
          </span>
        </button>
      </div>
      <div
        class="component"
      >
        <button
          class="component secondary m component secondary button secondary digit"
          type="button"
        >
          <span
            class="text stretchText"
          >
            5
          </span>
        </button>
      </div>
      <div
        class="component"
      >
        <button
          class="component secondary m component secondary button secondary digit"
          type="button"
        >
          <span
            class="text stretchText"
          >
            6
          </span>
        </button>
      </div>
      <div
        class="component"
      >
        <button
          class="component secondary m component secondary button secondary digit"
          type="button"
        >
          <span
            class="text stretchText"
          >
            7
          </span>
        </button>
      </div>
      <div
        class="component"
      >
        <button
          class="component secondary m component secondary button secondary digit"
          type="button"
        >
          <span
            class="text stretchText"
          >
            8
          </span>
        </button>
      </div>
      <div
        class="component"
      >
        <button
          class="component secondary m component secondary button secondary digit"
          type="button"
        >
          <span
            class="text stretchText"
          >
            9
          </span>
        </button>
      </div>
      <div />
      <div
        class="component"
      >
        <button
          class="component secondary m component secondary iconOnly button secondary digit"
          type="button"
        >
          0
        </button>
      </div>
      <div
        class="component"
      >
        <button
          class="component ghost m component ghost button ghost"
          type="button"
        >
          <span
            class="text stretchText"
          >
            <svg
              fill="currentColor"
              focusable="false"
              height="48"
              role="img"
              viewBox="0 0 48 48"
              width="48"
            >
              <path
                d="M27 22.221l-4.007-4.007-1.765 1.832 3.991 3.962-3.994 3.963 1.77 1.81L27 25.777l4.004 4.004 1.771-1.81-3.996-3.965L32.778 20 31 18.22l-4 4z"
              />
              <path
                clip-rule="evenodd"
                d="M19.922 11.45a2.99 2.99 0 00-2.132.888l-9.457 9.519a3.022 3.022 0 00-.883 2.142c0 .808.314 1.57.881 2.141l9.46 9.52a2.99 2.99 0 002.131.89H36c2.61 0 4.55-1.922 4.55-4.55V16c0-2.63-1.94-4.55-4.55-4.55H19.922zm-.642 2.368a.889.889 0 01.642-.269H36c.749 0 1.355.26 1.772.678.417.416.678 1.022.678 1.772v16c0 .75-.26 1.356-.678 1.773-.417.417-1.023.677-1.772.677H19.922a.89.89 0 01-.642-.268L9.82 24.66a.921.921 0 01-.27-.662c0-.254.094-.484.272-.662l9.457-9.52z"
                fill-rule="evenodd"
              />
            </svg>
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`PassCode Snapshot tests should match snapshot with unknown code length 1`] = `
<div>
  <div
    class="component"
  >
    <div
      class="error"
    />
    <div
      class="gap vertical"
      data-gap-size="m"
    />
    <div
      class="component"
    />
    <div
      class="gap vertical"
      data-gap-size="4xl"
    />
    <div
      class="component"
    >
      <div
        class="component"
      >
        <button
          class="component secondary m component secondary button secondary digit"
          type="button"
        >
          <span
            class="text stretchText"
          >
            1
          </span>
        </button>
      </div>
      <div
        class="component"
      >
        <button
          class="component secondary m component secondary button secondary digit"
          type="button"
        >
          <span
            class="text stretchText"
          >
            2
          </span>
        </button>
      </div>
      <div
        class="component"
      >
        <button
          class="component secondary m component secondary button secondary digit"
          type="button"
        >
          <span
            class="text stretchText"
          >
            3
          </span>
        </button>
      </div>
      <div
        class="component"
      >
        <button
          class="component secondary m component secondary button secondary digit"
          type="button"
        >
          <span
            class="text stretchText"
          >
            4
          </span>
        </button>
      </div>
      <div
        class="component"
      >
        <button
          class="component secondary m component secondary button secondary digit"
          type="button"
        >
          <span
            class="text stretchText"
          >
            5
          </span>
        </button>
      </div>
      <div
        class="component"
      >
        <button
          class="component secondary m component secondary button secondary digit"
          type="button"
        >
          <span
            class="text stretchText"
          >
            6
          </span>
        </button>
      </div>
      <div
        class="component"
      >
        <button
          class="component secondary m component secondary button secondary digit"
          type="button"
        >
          <span
            class="text stretchText"
          >
            7
          </span>
        </button>
      </div>
      <div
        class="component"
      >
        <button
          class="component secondary m component secondary button secondary digit"
          type="button"
        >
          <span
            class="text stretchText"
          >
            8
          </span>
        </button>
      </div>
      <div
        class="component"
      >
        <button
          class="component secondary m component secondary button secondary digit"
          type="button"
        >
          <span
            class="text stretchText"
          >
            9
          </span>
        </button>
      </div>
      <div />
      <div
        class="component"
      >
        <button
          class="component secondary m component secondary iconOnly button secondary digit"
          type="button"
        >
          0
        </button>
      </div>
      <div />
    </div>
  </div>
</div>
`;
